lib/graphql/static_validation/rules/fields_will_merge.rb

```diff
@@ -326,20 +327,19 @@ def find_fields_and_fragments(selections, owner_type:, parents:, fields:, fragme
         [fields, fragment_spreads]
       end
 
-      def possible_arguments(field1, field2)
+      def same_arguments?(field1, field2)
         # Check for incompatible / non-identical arguments on this node:
-        [field1, field2].map do |n|
-          if n.arguments.any?
-            serialized_args = {}
-            n.arguments.each do |a|
-              arg_value = a.value
-              serialized_args[a.name] = serialize_arg(arg_value)
-            end
-            serialized_args
-          else
-            NO_ARGS
-          end
-        end.uniq
+        arguments1 = field1.arguments
+        arguments2 = field2.arguments
+
+        return false if arguments1.length != arguments2.length
+
+        arguments1.all? do |argument1|
+          argument2 = arguments2.find { |argument| argument.name == argument1.name }
+          return false if argument2.nil?
+
+          serialize_arg(argument1.value) == serialize_arg(argument2.value)
+        end
       end
 
       def serialize_arg(arg_value)
```

Comment:
Not exactly in the scope of this PR, but this made me remember it might be better to use `GraphQL::Language::Printer` here since it already has this logic. @rmosolgo should that be used throughout instead of `GraphQL::Language.serialize` and `arg_value.to_query_string`?

Reply:
IMO the current code is _fine_. `.to_query_string` calls `Printer` under the hood: 

https://github.com/rmosolgo/graphql-ruby/blob/9cd0062330e225c90e6e4162e09c99384bf66a68/lib/graphql/language/nodes.rb#L76-L82

And `.serialize` is what `Printer` uses anyway: 

https://github.com/rmosolgo/graphql-ruby/blob/9cd0062330e225c90e6e4162e09c99384bf66a68/lib/graphql/language/printer.rb#L345-L351

So, practically speaking, I'm not sure what we'd gain from a refactor. (This method doesn't handle `Hash`es, but I guess it knows that it never receives a Hash? It would be nice if expected input classes were explicitly named by `when ...` and `else` raised an error, but that's not the end of the world.)